website/identifiers/clients/base.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+import requests
+
+from framework.exceptions import HTTPError
+from website import settings
+
+
+class AbstractIndentifierClient(object):
+
+    def __init__(self, base_url, prefix):
+        self.base_url = base_url
+        self.prefix = prefix
+
+    def build_metadata(self, object):
+        """ Build the metadata object used to register the object
+        with the specified client"""
+        raise NotImplementedError()
+
+    def build_doi(self, object):
```

Comment:
This also gets overriden by CrossRefClient. Maybe make this an abstract method (i.e. raise NotImplementedError)?